### DISCUSSION TITLE
Very important question! Problem with buy method.

### DISCUSSION THREAD

**Discussion by raoden1** [s4w3d0ff/python-poloniex#185] [Q&A] [answered]
Hi I have a question about buy method. Recently I received successful message from Poloniex
I run once buy order method with option immediateOrCancel.
I get two trades insted expected one.

 `{'orderNumber': '269159954730', 'amountUnfilled': '0.00000000', 'resultingTrades': [{'total': '0.00323034', 'tradeID': '16775174', 'amount': '0.10364748', 'rate': '0.03116670', 'date': '2018-04-22 10:59:11', 'type': 'buy'}, {'total': '0.05271018', 'tradeID': '16775175', 'amount': '1.68942915', 'rate': '0.03120000', 'date': '2018-04-22 10:59:11', 'type': 'buy'}]}`

Could someone help me with that or say why it's happend?

**Comment by wpank** [comment, marked as answer]
This is because your buy order consumed two different sell orders that were on the order book, resulting in two different trades. 

This is just the way that Poloniex handles it in their consumer api.
